2017/presentations/with_python_security/text.md

```diff
@@ -0,0 +1,519 @@
+An even better source of how an application should work are behavioural or integration tests.
+
+
+#### 2. Identification and quantification of threats
+
+There are few frameworks that we can work with, such as STRIDE
+(which stands for Spoofing identity, Tampering with data, Repudiation, Information disclosure, Denial of service, Elevation of privilege)
+and ASF (Application Security Frame), both of which should give us reliable information regarding:
+* auditing and logging
+    * auditing is used to answer the question "Who did what?" and possibly why.
+    * logging is more focused on what's happening.
+* authentication and authorization
+    * authentication is the a process of ascertaining that somebody really is who he or she claims to be
+    * authorization is a process to determine who is allowed to do what
+* configuration management – so how and where do we store configs
+* data validation and protection in storage and transit
+    * where do we store the data
+    * where do we validate it before recording
+        * backend side
+        * both backend and frontend side
+        * only frontend (is highly not advised)
+    * protection in transit
+        1. basic approach: have all communication over TLS, so encrypted
+        2. advanced approach: have the data additionally encrypted before sending it, which is becoming the corporate level security standard
+        3. expert approach: send it over a dedicated VPN tunnel, government level security
+* exception management – so how do we track exception occurrence and what are the procedures to handle them
+
+Now we just need to measure the severity of the threats we have.
+We can try approaching that by ourselves, determining which assets are the most important for us or, for example, use CVSS.
+CVSS, or Common Vulnerability Scoring System, is based on factors such as:
+* Attack vector
+* Attack complexity (hard to measure)
+* Privileges
+* User interaction
+* Scope
+* Confidentiality
+* Integrity
+* availability
+All of the above together can give a reference rating, but if one of the parameters is hard to define,
+like in the case of attack complexity, try both and either calculate an average or leave it as a range.
+It is important to remember that this is just a reference point, not an oracle for what should be done.
+As mentioned before, having use case UMLs or even better abuse case UMLs may come in handy here.
+
+#### 3. Addressing
+
+We can address issues in 4 ways, by:
+1. Completely removing the threat
+2. Reducing/mitigating the threat
+3. Acknowledging it and doing nothing
+4. Pretending there is no issue
+
+Obviously, the best options is to remove the threat, but sometimes it's either impossible or the costs of removing it are too high. In such cases we can try to mitigate it.
+Taking the risk by leaving it as is or marking it as "address later", where it does not affect our business nor customers,
```

Comment:
"Taking the risk [...] cats." - to jest długie zdanie i mimo wielokrotnego przeczytania nie zrozumiałem go dobrze. Warto byłoby je podzielić na kilka krótszych